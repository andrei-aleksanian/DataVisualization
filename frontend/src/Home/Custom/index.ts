export { default } from './Custom';
export * from './Custom';
