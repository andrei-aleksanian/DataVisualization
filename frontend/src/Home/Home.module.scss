.index {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-template-rows: repeat(12, 1fr);
  width: 100vw;
  height: 100vh;
  background: rgb(2, 0, 36);
  background: linear-gradient(135deg, rgba(2, 0, 36, 1) 0%, rgba(80, 20, 52, 1) 85%);
  color: var(--color-text);
}

.Container {
  grid-column-start: 2;
  grid-column-end: 9;
  grid-row-start: 3;
  grid-row-end: 8;
}

.Container * {
  margin: 1.5rem;
}

.Container h1 {
  font-size: 5rem;
  margin: 0 1.5rem;
}

.Container h2 {
  font-size: 2.2rem;
  margin: 0 1.5rem;
}

.Container p {
  font-size: 1.3rem;
  line-height: 1.35em;
  margin-bottom: 0;
}

.LinkWrapper {
  display: flex;
  margin: 0 1.5rem;
}

.LinkWrapper * {
  margin: 1.5rem 1rem 1.5rem 0;
}
