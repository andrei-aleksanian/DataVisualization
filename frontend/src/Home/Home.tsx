import { LinkHero } from 'Components/Link';
import classes from './Home.module.scss';

export const TEXT_LINK_LIBRARY = 'See an Example';
export const TEXT_LINK_CUSTOM_DATA = 'Visualize Your Data';

export const TEXT_H1 = 'COVA/ANGEL Visualization';
export const TEXT_P =
  "Dimensionality reduction for cohort data visualization is to plot a high-dimensional dataset in 2D,3D spaces to help analysts identify data properties and gain insights visually. COVA and ANGEL are two novel data visualization algorithms that provide a simple but effective way to improve cohort data visualization performance comprehensively. This demo website illustrates examples embedded by these two algorithms under different hyper-parameter settings and supports users in visualizing their datasets. In addition, the fast-ANGEL version is applied to the users' custom datasets. The current restriction of input data format is '.mat'.";
const Home = () => {
  return (
    <div className={classes.index}>
      <div className={classes.Container}>
        <h1>{TEXT_H1}</h1>
        <p>{TEXT_P}</p>
        <div className={classes.LinkWrapper}>
          <LinkHero text={TEXT_LINK_LIBRARY} link="/examples" />
          <LinkHero text={TEXT_LINK_CUSTOM_DATA} link="/your-data" />
        </div>
      </div>
    </div>
  );
};

export default Home;
