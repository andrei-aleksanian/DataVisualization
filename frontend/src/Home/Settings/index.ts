export { default } from './Settings';
export * from './Settings';
export * from './components/ANGEL';
export * from './components/COVA';
