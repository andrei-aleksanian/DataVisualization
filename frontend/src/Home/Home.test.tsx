import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import Home from './Home';

test('Settings matches snapshot', async () => {
  const { asFragment } = render(<Home />, { wrapper: MemoryRouter });
  expect(asFragment()).toMatchSnapshot();
});
