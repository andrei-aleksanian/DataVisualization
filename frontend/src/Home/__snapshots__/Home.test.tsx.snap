// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="index"
  >
    <div
      class="Container"
    >
      <h1>
        COVA / ANGEL
      </h1>
      <h2>
        Cohort Data Visualization
      </h2>
      <p>
        Dimensionality reduction for cohort data visualization is to plot a high-dimensional dataset in 2D,3D spaces to help analysts identify data properties and gain insights visually. COVA and ANGEL are two novel data visualization algorithms that provide a simple but effective way to improve cohort data visualization performance comprehensively. This demo website illustrates examples embedded by these two algorithms under different hyper-parameter settings and supports users in visualizing their datasets.
      </p>
      <div
        class="LinkWrapper"
      >
        <a
          class="index Action"
          href="/examples"
        >
          See an Example
        </a>
        <a
          class="index Action"
          href="/your-data"
        >
          Visualize Your Data
        </a>
        <a
          class="index Action external"
          href="http://docs.see-your-embeddings.com/"
          rel="noreferrer"
          target="_blank"
        >
          See Docs
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
